crates/matrix-sdk-ui/src/timeline/controller/mod.rs

```diff
@@ -1072,7 +1072,11 @@ impl<P: RoomDataProvider, D: Decryptor> TimelineController<P, D> {
                 warn!("We looked for a local item, but it transitioned as remote??");
                 return false;
             };
-            prev_local_item.with_send_state(EventSendState::NotSentYet)
+            // If the local echo had an upload progress, retain it.
+            let progress = as_variant!(&prev_local_item.send_state,
+                EventSendState::NotSentYet { progress } => progress.clone())
+            .flatten();
```

Comment:
Let's let the code breath a bit, and simplify the `as_variant!` to do as little as possible:

```suggestion

            // If the local echo had an upload progress, retain it.
            let progress = as_variant!(&prev_local_item.send_state, EventSendState::NotSentYet)
              .and_then(|progress| progress.cloned());
            
```

Reply:
ah crap. Thanks for trying though!